### PR TITLE
chore: add upgrade job

### PR DESCRIPTION
**Pull request by ricolin**
Relate-to: https://github.com/vexxhost/atmosphere/issues/1068
Relate-to: https://github.com/vexxhost/atmosphere/issues/1282

Depends-On: https://github.com/vexxhost/atmosphere/pull/1408
Depends-On: https://github.com/vexxhost/atmosphere/pull/1409

### CODE REVIEW

**Comment by mnaser** [comment]
@ricolin does it make sense to branch 2024.1 so we can make upgrade job from 2024.1 to main since thats a more supported upgrade route?

The other thing I wanted to bring up is what do you think of introducing a new upgrade pipeline, because if we run upgrade jobs on every commit, we doubled or CI time potentially (which is already kinda long tbh)...

**Comment by ricolin** [comment]
> @ricolin does it make sense to branch 2024.1 so we can make upgrade job from 2024.1 to main since thats a more supported upgrade route?
I trust your experience with upgrade:) and avoid cross multi-version upgrade sounds like a good plan too.

> 
> The other thing I wanted to bring up is what do you think of introducing a new upgrade pipeline, because if we run upgrade jobs on every commit, we doubled or CI time potentially (which is already kinda long tbh)...

Agree, +1 on a upgrade pipeline with comment_filter so we can do verify when we plan to before release or make it a perodic job.

File: roles/keycloak/tasks/main.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@
       - version
   register: mysql_ready
   until: mysql_ready is not failed
-  retries: 60
```
Comment (mnaser) [inline roles/keycloak/tasks/main.yml:37]: can you make this a separate change please?

**Review by mnaser — CHANGES_REQUESTED** [review]
I'm going to spend a bit of time thinking how we can accomplish this in a way where we have a separate pipeline probably or a periodic job.

File: roles/keycloak/tasks/main.yml (hunk 1 above)
Comment (ricolin) [reply roles/keycloak/tasks/main.yml]: https://github.com/vexxhost/atmosphere/pull/1409

File: zuul.d/jobs.yaml [hunk 2]
```diff
@@ -84,3 +84,24 @@
...
+    name: atmosphere-molecule-aio-upgrade-base
+    run: zuul.d/playbooks/upgrade/run-upgrade.yml
+    abstract: true
+    timeout: 10800
+    parent: atmosphere-molecule-aio-full
+    vars:
+      from_branch: "stable/2023.2"
+
+- job:
+    name: atmosphere-molecule-aio-upgrade-ovs
```
Comment (mnaser) [inline zuul.d/jobs.yaml:98]: For consistency, please rename the job to use `openvswitch` instead of `ovs`.

**Review by mnaser — CHANGES_REQUESTED** [review]

File: zuul.d/jobs.yaml (hunk 2 above)
Comment (ricolin) [reply zuul.d/jobs.yaml]: done

**Comment by ricolin** [comment]
recheck

**Review by mnaser — APPROVED** [review]

**Pull request merged**